src/core/main.c

```diff
@@ -1982,6 +1983,12 @@ static int do_reexecute(
                         log_error_errno(r, "Failed to switch root, trying to continue: %m");
         }
 
+        r = 0;
+        if (getpid_cached() != 1)
+                r = capability_ambient_set_apply(capability_ambient_set, /* also_inherit= */ false);
+        if (r < 0)
+                return log_error_errno(errno, "Failed to apply the starting ambient set: %m");
```

Comment:
why condition this on pid 1? seems reasonably safe to always do, no?

Reply:
pid 1 has an empty ambient set by default and does not modify it anywhere for itself so calling `capability_ambient_set_apply()` is no-op. A call to `capability_ambient_set_apply()` is not time consuming but makes little sense for pid 1. IMHO it also feels more appropriate to check that condition as it is checked [here](https://github.com/systemd/systemd/blob/ce2aade08c9d7b67514550f263a2148700892f3e/src/core/main.c#L3136) and to restore the ambient set only when it was dropped. But if you prefer to drop that first `if` the code should work fine.